### PR TITLE
Fix crash when saving image URLs by aggressively rejecting nil (NSNul…

### PR DESCRIPTION
**Pull request by coreyfloyd**
…l) URLs

- Reject nil URLs when retrieving imageInfoForTitle from the datastore
- Check for the proper class (MWKImageInfo) before mapping to a URL (previously a KVO key value coding call on an unknown class)
- Check for the proper class (NSURL) before mapping to a scheme less URL (previously assumed it was an NSURL and crashed)

- Additionally refactor both scheme removal code paths into one (in NSURL category)

### CODE REVIEW

File: Wikipedia/Code/MWKArticle.m [hunk 1]
```diff
@@ -431,14 +431,12 @@ - (NSString*)debugDescription {
...
 
--(NSArray<NSURL*>*)schemelessURLs:(NSArray<NSURL*>*)urls {
-    return [urls bk_map:^NSURL*(NSURL* url){
-        NSString* urlStr = url.absoluteString;
-        NSRange dividerRange = [urlStr rangeOfString:@"://"];
-        if (dividerRange.location != NSNotFound) {
-            return [NSURL URLWithString:[urlStr substringFromIndex:dividerRange.location + 1]];
+-(NSArray<NSURL*>*)schemelessURLsRejectingNilURLs:(NSArray<NSURL*>*)urls {
+    return [urls wmf_mapAndRejectNil:^NSURL*(NSURL* url){
+        if([url isKindOfClass:[NSURL class]]){
```
Comment (montehurd) [inline Wikipedia/Code/MWKArticle.m]: Is this class check redundant because urls is already `NSArray<NSURL*>*`?

**Comment by joewalsh** [comment]
👍 pending monte's additions

File: Wikipedia/Code/MWKArticle.m (hunk 1 above)
Comment (coreyfloyd) [inline Wikipedia/Code/MWKArticle.m]: Already talked about in hangout but for posterity:

It can't be guaranteed - it is only a hint to the compiler

**Comment by montehurd** [comment]
@coreyfloyd 👍 if my last commits look ok to you

**Comment by joewalsh** [comment]
👍

**Pull request merged**